app/components/post_body_additional_content/index.js

```diff
@@ -45,7 +46,14 @@ function makeMapStateToProps() {
 
     return function mapStateToProps(state, ownProps) {
         const config = getConfig(state);
-        const link = getFirstLink(ownProps.message);
+        const {metadata} = ownProps;
+        let link;
+        if (metadata) {
+            const embedData = getEmbedFromMetadata(metadata);
+            link = embedData && embedData.url;
```

Comment:
what if `embedData` or  `embedData.url` are null? should this be something like
```suggestion
            link = embedData?.url || getFirstLink(ownProps.message);
```

Reply:
As long as we have metadata, the server would've told us whether or not there's a link in it. We don't want to have the mobile app override that since that'll result in scroll pop if it does find something that the server can't

Reply:
One other thing is that the url from here matches to the openGraph data in the store by the URL so, using different ones(i.e getFirstLink) can have issues.  Logic here for completely trusting server if metadata exists works on webapp so far. So i think this is good